src/adstaylorinteg.jl

```diff
@@ -0,0 +1,326 @@
+    # Size per variable
+    M = size_per_variable(P)
+    # Variable with maximum error
+    return argmax(M)[1]
+end
+
+function splitdirection(P::AbstractVector{TaylorN{T}}) where {T <: Real}
+    # Size per variable
+    M = size_per_variable(P)
+    # Variable with maximum error
+    return argmax(M)[1]
+end
+
+"""
+    adsnorm(P::TaylorN{T}) where {T <: Real}
+
+Return the truncation error of `P`.
+"""
+function adsnorm(P::TaylorN{T}) where {T <: Real}
+    # Jet transport order
+    varorder = P.order
+    # Absolute sum per order
+    ys = Vector{T}(undef, varorder+1)
+
+    for i in eachindex(ys)
+        ys[i] = sum(abs, P.coeffs[i].coeffs)
+    end
+    # Initial parameters
+    p0 = ones(T, 2)
+    # Orders
+    xs = 0:varorder
+    # Non zero coefficients
+    idxs = findall(!iszero, ys)
+    # Fit exponential model
+    fit = curve_fit(exp_model!, exp_model_jacobian!, view(xs, idxs),
+                        view(ys, idxs), p0; inplace = true)
+    # Estimate next order coefficient
+    return exp_model(varorder+1, fit.param)
+end
+
+# Split node's domain in half
+# See section 3 of https://doi.org/10.1007/s10569-015-9618-3
+function split(node::ADSBinaryNode{N, M, T}, x::SVector{M, TaylorN{T}},
+               p::SVector{M, Taylor1{TaylorN{T}}}) where {N, M, T <: Real}
+    # Split direction
+    j = splitdirection(x)
+    # Split domain
+    s1, s2 = split(node.s, j)
+    # Jet transport variables
+    v_1, v_2 = get_variables(), get_variables()
+    # Shift expansion point
+    r = getroot(node).s
+    v_1[j] = v_1[j]/2 + r.lo[j]/2
+    v_2[j] = v_2[j]/2 + r.hi[j]/2
+    # Taylor1 order
+    order = p[1].order
+    # Left half
+    x1 = SVector{M, TaylorN{T}}(x[i](v_1) for i in eachindex(x))
+    p1 = SVector{M, Taylor1{TaylorN{T}}}(
+        Taylor1( map(z -> z(v_1), p[i].coeffs), order ) for i in eachindex(p)
+    )
+    # Right half
+    x2 = SVector{M, TaylorN{T}}(x[i](v_2) for i in eachindex(x))
+    p2 = SVector{M, Taylor1{TaylorN{T}}}(
+        Taylor1( map(z -> z(v_2), p[i].coeffs), order ) for i in eachindex(p)
+    )
+
+    return s1, x1, p1, s2, x2, p2
+end
+
+"""
+    split!(node::ADSBinaryNode{N, M, T}, p::SVector{M, Taylor1{TaylorN{T}}},
+           dt::T, nsplits::Int, maxsplits::Int, stol::T) where {N, M, T <: Real}
+
+Split `node` in half if any element of `p(dt)` has an `adsnorm` greater than `stol`.
+"""
+function split!(node::ADSBinaryNode{N, M, T}, p::SVector{M, Taylor1{TaylorN{T}}},
+                dt::T, nsplits::Int, maxsplits::Int, stol::T) where {N, M, T <: Real}
+    # Evaluate x at dt
+    x = _adseval(p, dt)
+    # Split criteria for each element of x
+    mask = adsnorm.(x)
+    # Split
+    if nsplits < maxsplits && any(mask .> stol)
+        # Split
+        s1, x1, p1, s2, x2, p2 = split(node, x, p)
+        # Left half
+        leftchild!(node, s1, node.t + dt, x1, p1)
+        # Right half
+        rightchild!(node, s2, node.t + dt, x2, p2)
```

Comment:
Related to my comment above, what I'm proposing is:
```suggestion
        # Split
        lchildnode, rchildnode = split(node, x, p, dt) # note the addition of `dt` to the `split` call
        # Left half
        leftchild!(node lchildnode) # call to proposed leftchild! method
        # Right half
        rightchild!(node, rchildnode) # call to proposed rightchild! method
```

Reply:
Thanks for the suggestion! Now `split` returns a tuple of `ADSBinaryNode`.